### ISSUE TITLE
Can't install extensions anymore

### ISSUE THREAD

**Issue by devodev** [docker/for-win#13084]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: B892056C-B25B-4D81-AF48-3891A00FCABF/20221120160158

### Actual behavior

After the latest update to Docker Desktop, I can't install extensions and receive the following error for every one of them:

```
Failed to install extension: the local image you are trying to install has been modified and is different from the published image joycelin79/newman-extension:0.0.7
Failed to install extension: the local image you are trying to install has been modified and is different from the published image docker/resource-usage-extension:1.0.2
...
```

### Expected behavior

I can install extensions.

### Information

  - Windows Version: `Version	10.0.19044 Build 19044`
  - Docker Desktop Version: `4.14.1 (91661)`
  - WSL2 or Hyper-V backend? `WSL2`
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: `No`

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`

```
"C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check
[2022-11-20T16:02:44.435462900Z][com.docker.diagnose.exe][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0002: does the bootloader have virtualization enabled?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0005: is the user in the docker-users group?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[PASS] DD0012: is the VM networking working?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?
No fatal errors detected.
```

### Steps to reproduce the behavior

I tried uninstalling the app completely, then rebooting and re-installing.

I also went into the `Troubleshoot` section and hit `Clean / Purge data` but without success.

I tried disabling beta features without success either.

### Workaround

It seems that this is caused by one or multiple corrupted files under `$HOME/.docker`.

Thanks to @SOLIJA for providing a simple fix:

> I fixed it exiting docker and by deleteing %userprofile%.docker folder then starting docker again.
> 
> If you don't want to delete that whole folder. I believe just deleting the Contexts folder inside .docker and restarting docker will fix it also.

**Comment by IeuanWalker** [comment]
Also getting this, i've just reinstalled windows on a new drive, was working fine before that

**Comment by ig-rudenko** [comment]
Same problem : (

**Comment by TjLuckyBastard** [comment]
Hi there!,
Same problems here.

**Comment by SOLIJA** [comment]
Getting the same error with the Volumes Backup and Share extension.

**Comment by groogiam** [comment]
I am getting the same error as well.

**Comment by SOLIJA** [comment]
I fixed it exiting docker and by deleteing %userprofile%\.docker folder then starting docker again.

**Comment by devodev** [comment]
> I fixed it exiting docker and by deleteing %userprofile%.docker folder then starting docker again.
> 
> If you don't want to delete that whole folder. I believe just deleting the Contexts folder inside .docker and restarting docker will fix it also.

Thank you, it worked!

Will add to the initial post :)

**Comment by groogiam** [comment]
This worked for me as well. Thanks

**Comment by meluhian** [comment]
> 

Thanks!! Worked :)

**Comment by AmbassadorLaZer** [comment]
> I fixed it exiting docker and by deleteing %userprofile%.docker folder then starting docker again.
> 
> If you don't want to delete that whole folder. I believe just deleting the Contexts folder inside .docker and restarting docker will fix it also.

Didn't work for me sadly.

**Comment by shanselman** [comment]
Hit this today as well. Deleting contexts folder fixed, it but this is a long standing bug

**Comment by thousandcity** [comment]
deleting the Contexts folder inside .docker and restarting works well!

**Comment by AmIBeingObtuse** [comment]
> I fixed it exiting docker and by deleteing %userprofile%.docker folder then starting docker again.
> 
> If you don't want to delete that whole folder. I believe just deleting the Contexts folder inside .docker and restarting docker will fix it also.

Worked for me thank you.

**Comment by rob-baldwin** [comment]
> > I fixed it exiting docker and by deleteing %userprofile%.docker folder then starting docker again.
> > If you don't want to delete that whole folder. I believe just deleting the Contexts folder inside .docker and restarting docker will fix it also.
> 
> Didn't work for me sadly.

This didn't work for me either, were you able to find any way to resolve the issue?

**Comment by rob-baldwin** [comment]
I finally found out that I must have installed azure iotedge at some point in the past, and it has it's own version of docker that was conflicting, once I remove iotedge then everything started to work as expected.

**Comment by chiptus** [comment]
> I fixed it exiting docker and by deleteing %userprofile%/.docker folder then starting docker again.
> 
> If you don't want to delete that whole folder. I believe just deleting the Contexts folder inside .docker and restarting docker will fix it also.

it also fixed an issue of uninstalling extensions. thanks!

**Comment by benja-M-1** [comment]
As for https://github.com/docker/for-mac/issues/6695, a fix was expected in 4.18 which has been released a while ago. I am closing this issue. Feel free to reopen if it still happens.
